### PR TITLE
Initialize readiness package

### PR DESCRIPTION
**Pull request by avi-08**
### What this PR does / why we need it
- Initializes readiness package with controller deployment and rbac manifests.
- Updates `docs/packages/dev-workflow.md` with build and package instructions

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

- Built and published package using build-tooling.
```
OCI_REGISTRY="****" make docker-build-all -f build-tooling.mk

OCI_REGISTRY="****" make docker-publish-all -f build-tooling.mk

OCI_REGISTRY="****" \
PACKAGE_NAME="readiness" \
PACKAGE_REPOSITORY="management" \
PACKAGE_VERSION="dev.0.0" \
PACKAGE_SUB_VERSION="v1" \
REGISTRY_USERNAME="****" \
REGISTRY_PASSWORD="****" \
REGISTRY_SERVER="****" \
make package-bundle-generate package-bundle-push -f build-tooling.mk
```

- Deployed the package on kind cluster successfully.
```
imgpkg pull -b readiness:dev.0.0_v1 -o readiness-pkg
ytt -f readiness-pkg/config/ | kbld -f - -f readiness-pkg/.imgpkg/images.yml | kubectl apply -f-
```

### Release note

```release-note

```

### CODE REVIEW

File: packages/readiness/Makefile
Comment (sathyanarays) [inline packages/readiness/Makefile:1]: Why is this file required?

File: packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinesses.yaml
Comment (sathyanarays) [inline packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinesses.yaml:1]: Do we need to copy the generated CRDs? Is it not possible to copy them during build time? Otherwise, we need to keep making some effort to maintain the consistency.

File: packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinessproviders.yaml
Comment (sathyanarays) [inline packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinessproviders.yaml:1]: Same as the above file

File: packages/readiness/bundle/config/upstream/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  name: tanzu-readiness-manager-sa
+  namespace: #@ data.values.namespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  annotations:
+    kapp.k14s.io/change-group: "readiness.core.tanzu.vmware.com/serviceaccount"
+  name: tanzu-readiness-manager-clusterrole
+rules:
```
Comment (sathyanarays) [inline packages/readiness/bundle/config/upstream/rbac.yaml:19]: We need to extend the rules so that it could read any resource of any type.

Comment (sathyanarays) [reply packages/readiness/bundle/config/upstream/rbac.yaml:19]: We can do that later though!

File: packages/readiness/bundle/config/upstream/tanzu-readiness-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tanzu-readiness-controller-manager
+  labels:
+    app: tanzu-readiness-manager
+  namespace: #@ data.values.namespace
+spec:
+  replicas: 3
```
Comment (sathyanarays) [inline packages/readiness/bundle/config/upstream/tanzu-readiness-manager.yaml:11]: We can start with 1 replica

File: packages/readiness/bundle/config/values.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#@data/values
+---
+namespace: default
+deployment:
+  hostNetwork: false
+  nodeSelector: null
+  tolerations: []
```
Comment (sathyanarays) [inline packages/readiness/bundle/config/values.yaml:7]: Are the deployment related parameters relevant to us now?

File: packages/readiness/bundle/config/upstream/rbac.yaml (hunk 1 above)
Comment (avi-08) [reply packages/readiness/bundle/config/upstream/rbac.yaml:19]: Yes, right now they only contain access rules specific to readiness resource. We can keep updating this list as we go.

File: packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinesses.yaml
Comment (avi-08) [reply packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinesses.yaml:1]: Its not manually copied. vendir takes care of maintaining the synchronization from repo contents. See `packages/readiness/vendir.yaml`

File: packages/readiness/bundle/config/upstream/tanzu-readiness-manager.yaml (hunk 2 above)
Comment (avi-08) [reply packages/readiness/bundle/config/upstream/tanzu-readiness-manager.yaml:11]: ok

File: packages/readiness/Makefile
Comment (avi-08) [reply packages/readiness/Makefile:1]: This was added as part of the [package template](https://github.com/vmware-tanzu/build-tooling-for-integrations/blob/main/docs/add-package.md).

File: packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinessproviders.yaml
Comment (avi-08) [reply packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinessproviders.yaml:1]: Its not manually copied. vendir takes care of maintaining the synchronization from repo contents. See packages/readiness/vendir.yaml

**Comment by avi-08** [comment]
> Can we add manual instructions for building and deploying readiness framework? May be in a MD file similar to the instructions given in "Describe testing done for PR" section?

Updated `docs/packages/dev-workflow.md`

File: packages/readiness/bundle/config/upstream/tanzu-readiness-manager.yaml (hunk 2 above)
Comment (avi-08) [reply packages/readiness/bundle/config/upstream/tanzu-readiness-manager.yaml]: Updated

File: packages/readiness/bundle/config/values.yaml (hunk 3 above)
Comment (avi-08) [reply packages/readiness/bundle/config/values.yaml:7]: These are optional parameters, consumers of the package can configure if required.

File: packages/readiness/bundle/config/upstream/rbac.yaml (hunk 1 above)
Comment (rajathagasthya) [reply packages/readiness/bundle/config/upstream/rbac.yaml:19]: What resources are we expecting to read here? Anything in the cluster?

Comment (rajathagasthya) [reply packages/readiness/bundle/config/upstream/rbac.yaml:19]: If it's "any" resource in the cluster, you should consider taking a service account as an input in the API instead of adding a `*` rule here, which is not best practice. [See how Capability API does this](https://github.com/vmware-tanzu/tanzu-framework/blob/6a32fdc564d967d9cd64d96a1af11aa55d54f860/apis/core/v1alpha2/capability_types.go#L13-L19).

This makes the user responsible for providing the right RBAC.

File: packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinesses.yaml
Comment (sathyanarays) [reply packages/readiness/bundle/config/upstream/crds/core/core.tanzu.vmware.com_readinesses.yaml:1]: Okay!

File: packages/readiness/bundle/config/upstream/rbac.yaml (hunk 1 above)
Comment (sathyanarays) [reply packages/readiness/bundle/config/upstream/rbac.yaml:19]: Thanks @rajathagasthya .. this makes sense!

File: packages/readiness/bundle/config/upstream/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    verbs:
+      - get
+      - patch
+      - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  annotations:
+    kapp.k14s.io/change-rule: "upsert after upserting readiness.core.tanzu.vmware.com/serviceaccount"
```
Comment (yharish991) [inline packages/readiness/bundle/config/upstream/rbac.yaml:51]: i think you also need to add this annotation to handle the package deletion case 
```  
 kapp.k14s.io/change-rule.1: "delete before deleting readiness.core.tanzu.vmware.comserviceaccount"
```

File: docs/packages/dev-workflow.md [hunk 5]
```diff
@@ -2,93 +2,120 @@
...
+      imgpkg pull -b ${OCI_REGISTRY}/readiness@sha256:1fb9f9c6f0c6ba1f995440885b02806551a79d9cef5b9c7c3d6f53a586facddd -o readiness-pkg
+   ```
+
+1. Install the package
+   ```shell
+      ytt -f readiness-pkg/config/ | kbld -f - -f readiness-pkg/.imgpkg/images.yml | kubectl apply -f-
+   ```
+
+#### To install on a cluster with `kapp-controller` installed in it.
 You can also use [management cluster](https://github.com/vmware-tanzu/tanzu-framework/blob/main/cmd/cli/plugin/managementcluster/README.md)
```
Comment (yharish991) [inline docs/packages/dev-workflow.md:119]: i think we can remove this instruction

File: docs/packages/dev-workflow.md [hunk 6]
```diff
@@ -2,93 +2,120 @@
...
+----
+
 Follow the below steps to test the artifacts that are generated in previous steps on your local cluster
 
-**Note**: A local cluster can be any k8s cluster with `kapp-controller` installed in it.
+> **Note**: A local cluster can be any k8s cluster. There are 2 ways to install the packages
+
+#### To install on a cluster without `kapp-controller` installed in it.
+
+1. Download the package bundle. The sha tag can be noted from the registry where package was published.
```
Comment (yharish991) [inline docs/packages/dev-workflow.md:108]: I think the instructions should be to create the PackageRepository CR first, which makes the packages available in the cluster, and then to install a particular package they are interested in by creating the PackageInstall CR

or

pull and extract the repo bundle, and create the Package CR and the PackageInstall CR

**Review by sathyanarays — APPROVED** [review]

File: docs/packages/dev-workflow.md (hunk 6 above)
Comment (avi-08) [reply docs/packages/dev-workflow.md]: Updated

File: docs/packages/dev-workflow.md (hunk 5 above)
Comment (avi-08) [reply docs/packages/dev-workflow.md]: Updated

File: packages/readiness/bundle/config/upstream/rbac.yaml (hunk 4 above)
Comment (avi-08) [reply packages/readiness/bundle/config/upstream/rbac.yaml:51]: Done

File: docs/packages/dev-workflow.md [hunk 7]
```diff
@@ -2,107 +2,175 @@
...
+      metadata:
+      name: tanzu-management-repo
+      spec:
+      fetch:
+         imgpkgBundle:
+            image: ${OCI_REGISTRY}/management@sha256:1fb9f9c6f0c6ba1f995440885b02806551a79d9cef5b9c7c3d6f53a586facddd
+   ```
+
+   ```bash
+      kapp deploy -a repo -f repo.yml -y
```
Comment (yharish991) [inline docs/packages/dev-workflow.md:150]: nit: let's keep the extension consistent as `yaml`

**Review by yharish991 — APPROVED** [review]
LGTM!! will you be updating the package repository name in a follow-up pr?

**Comment by avi-08** [comment]
> LGTM!! will you be updating the package repository name in a follow-up pr?

Since this branch is behind `feature/runtime-core` by a few commits, I'll make this change with a rebase commit in next PR.

File: docs/packages/dev-workflow.md (hunk 7 above)
Comment (avi-08) [reply docs/packages/dev-workflow.md]: Updated
